### PR TITLE
plugins/lsp: add fortls

### PR DESCRIPTION
**Pull request by loicreynier**
Add `fortls` LSP plugin. 

This plugin features numerous options, some of which are experimental. At the moment, I haven't implemented any `settingsOptions` because I'm not sure it's necessary given that the LSP is entirely configured by its command line arguments (it could then be configured through `fortls.cmd = [ ... ]`. Plus, it can be configured via a configuration file. Adding the options in nixvim would add significant maintenance work. However, I'm not closed to the idea of adding the options if you think it's necessary (as mentioned in the contribution guideline).

As for the default options, I've chosen those selected by [`nvim-lspconfig`](https://github.com/neovim/nvim-lspconfig/blob/master/lua/lspconfig/server_configurations/fortls.lua) (except for the initialization notification one

### CODE REVIEW

File: plugins/lsp/language-servers/default.nix [hunk 1]
```diff
@@ -210,6 +210,16 @@ with lib; let
       package = pkgs.elixir-ls;
       cmd = cfg: ["${cfg.package}/bin/elixir-ls"];
     }
+    {
+      name = "fortls";
+      package = pkgs.fortls;
```
Comment (GaetanLepage) [inline plugins/lsp/language-servers/default.nix:215]:
```suggestion
```
As the package has the same name as the LSP, you can omit this line.

File: plugins/lsp/language-servers/default.nix [hunk 2]
```diff
@@ -210,6 +210,15 @@ with lib; let
...
       cmd = cfg: ["${cfg.package}/bin/elixir-ls"];
     }
+    {
+      name = "fortls";
+      cmd = cfg: [
+        "${cfg.package}/bin/fortls"
+        "--hover_signature"
+        "--hover_language=fortran"
+        "--use_signature_help"
+      ];
```
Comment (GaetanLepage) [inline plugins/lsp/language-servers/default.nix:220]:
```suggestion
```

As this is the default `nvim-lspconfig` value for `cmd`, I am not sure that you have to add this...

**Comment by GaetanLepage** [comment]
Could you please add a `description`:
```nix
description = "fortls for Fortran";
```

File: plugins/lsp/language-servers/default.nix (hunk 2 above)
Comment (loicreynier) [reply plugins/lsp/language-servers/default.nix:220]: It's not exactly the default command, I remove the startup notification.

Comment (GaetanLepage) [reply plugins/lsp/language-servers/default.nix:221]: Ok fine !

**Review by GaetanLepage — APPROVED** [review]

**Review by traxys — APPROVED** [review]

**Pull request merged**